scalafix-cli/src/main/scala/scalafix/internal/jgit/JGitBlame.scala

```diff
@@ -0,0 +1,89 @@
+package scalafix.internal.jgit
+
+import scalafix.lint.{RuleDiagnostic, Diagnostic}
+
+import scala.meta.Input
+
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.blame.{BlameGenerator, BlameResult}
+import org.eclipse.jgit.lib.Constants
+
+import java.util.Date
+import java.text.SimpleDateFormat
+import java.nio.file.{Path, Paths}
+
+import org.eclipse.jgit.revwalk.RevCommit
+
+class JGitBlame(workingDir: Path, diffBase: Option[String]) {
+  private val builder = new FileRepositoryBuilder()
+  private val repository =
+    builder.readEnvironment().setWorkTree(workingDir.toFile).build()
+  private val dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm")
+
+  def apply(diagnostics: List[RuleDiagnostic]): List[RuleDiagnostic] = {
```

Comment:
You already have access to the nio.Path file in `unsafeHandleFile`
```
  def unsafeHandleFile(args: ValidatedArgs, file: AbsolutePath): ExitStatus = {
```

That would avoid the `val path = ...` match below.